backend/internal/controller/http/helpers.go

```diff
@@ -0,0 +1,29 @@
+package http
+
+import (
+	"github.com/MAXXXIMUS-tropical-milkshake/MAXXXIMUS_Calculator/pkg/logger"
+	"github.com/gofiber/fiber/v2"
+	"github.com/MAXXXIMUS-tropical-milkshake/MAXXXIMUS_Calculator/internal/controller/http/model/validator"
+	"fmt"
+	"github.com/MAXXXIMUS-tropical-milkshake/MAXXXIMUS_Calculator/internal/controller/http/model"
+)
+
+func parseQueryAndValidate(ctx *fiber.Ctx, formValidator validator.FormValidatorService, data interface{}) (fiberError, parseOrValidationError error) {
+	if err := ctx.QueryParser(data); err != nil {
+		logger.Log().Error(ctx.UserContext(), err.Error())
+		return ctx.Status(fiber.StatusBadRequest).JSON(model.ErrorResponse(err.Error())), err
+	}
+
+	return validate(ctx, formValidator, data)
+}
+
+func validate(ctx *fiber.Ctx, formValidator validator.FormValidatorService, data interface{}) (fiberError, parseOrValidationError error) {
+	errs := formValidator.Validate(data)
+	if len(errs) > 0 {
```

Comment:
matter of opinion, but `if len(errs) == 0 { return nil, nil } ` seems less nested